File: app/src/main/java/com/example/tetrisapp/model/game/configuration/PieceConfigurations.java
package com.example.tetrisapp.model.game.configuration;

import com.example.tetrisapp.interfaces.PieceConfiguration;

public enum PieceConfigurations {
    DEFAULT(new PieceConfigurationDefault()),
    GLASS(new PieceConfigurationGlass()),
    BRICK(new PieceConfigurationBrick()),
    PORTAL_CUBE(new PieceConfigurationPortalCube());

    private final PieceConfiguration configuration;

    PieceConfigurations(PieceConfiguration configuration) {
        this.configuration = configuration;
    }

    public PieceConfiguration getConfiguration() {
        return this.configuration;
    }
}

File: app/src/main/java/com/example/tetrisapp/model/game/configuration/PieceConfigurations.kt
package com.example.tetrisapp.model.game.configuration

import com.example.tetrisapp.interfaces.PieceConfiguration

enum class PieceConfigurations(configuration: PieceConfiguration) {
    DEFAULT(PieceConfigurationDefault()),
    GLASS(PieceConfigurationGlass()),
    BRICK(PieceConfigurationBrick()),
    PORTAL_CUBE(PieceConfigurationPortalCube());

    val configuration: PieceConfiguration

    init {
        this.configuration = configuration
    }
}